servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -183,6 +192,55 @@ final case class NonBlockingServletIo[F[_]: Async](chunkSize: Int) extends Servl
       }
     }
 
+  /* The queue implementation is influenced by ideas in jetty4s
+   * https://github.com/IndiscriminateCoding/jetty4s/blob/0.0.10/server/src/main/scala/jetty4s/server/HttpResourceHandler.scala
+   */
+  override def requestBody(
+      servletRequest: HttpServletRequest,
+      dispatcher: Dispatcher[F],
+  ): Stream[F, Byte] = {
+    sealed trait Read
+    final case class Bytes(chunk: Chunk[Byte]) extends Read
+    case object End extends Read
+    final case class Error(t: Throwable) extends Read
+
+    Stream.eval(F.delay(servletRequest.getInputStream)).flatMap { in =>
+      Stream.eval(Queue.synchronous[F, Read]).flatMap { q =>
+        val readBody = Stream.exec(F.delay(in.setReadListener(new ReadListener {
+          def onDataAvailable(): Unit = {
+            def go: F[Unit] =
+              F.delay(new Array[Byte](chunkSize)).flatMap { buff =>
+                F.delay(in.read(buff)).flatMap {
+                  case len if len >= 0 =>
+                    q.offer(Bytes(Chunk.array(buff, 0, len))) >>
```

Comment:
This is simpler than jetty4s, and avoids copying bytes from partial chunks.  But it wastes (chunkSize - len) bytes of memory per chunk.  I think I like what jetty4s did better.